Legislation as Code details (#219)
ServiceInnovationLab/serviceinnovationlab.github.io: _services_projects/legislation-as-code.md

quote:
media: "[Embracing Innovation in Government - Global Trends 2019](https://trends.oecd-opsi.org/)"
---

## Work to date

Our rules can be viewed at <https://rules.nz>

### Birth of a Child (or welcoming a child into your whanau)

Encoding entitlements related to the birth of a child (or welcoming a child into your whanau) into re-usable open legislation as code. This component was then used within [Smart Start - financial help](https://smartstart.services.govt.nz/financial-help).

#### Child related entitlements encoded

Social Security Act 2018

* Accommodation Supplement
* Childcare Subsidy
* Child Disability Allowance
* Community Services Card
* Home Help
* Job Seeker Support
* Orphan's Benefit
* Sole Parent Support
* Supported Living Payment
* Unsupported Child
* Young Parent Payment

New Zealand Superannuation and Retirement Income Act 2001

* Superannuation

Income Tax Act 2007

* Working for Families (various)
* Best Start

Parental Leave and Employment Protection Act 1987

* Paid Parental Leave

### Citizenship by Grant

> A complex forumla from within the Citizenship Act, used to calculated if applicant has been in the country enough days to satisfy a clause. This was encoded into Open Fisca, along with some related sections of Immigration Act.

Reviewer: ```suggestion
A complex forumla from within the Citizenship Act, used to calculate if an applicant has been in the country enough days to satisfy a clause. This was encoded into OpenFisca, along with some related sections of Immigration Act.
```

quote:
media: "[Embracing Innovation in Government - Global Trends 2019](https://trends.oecd-opsi.org/)"
---

> ## Work to date

Author: ```suggestion
## Work dsjfkdajfklda
```

> 

Author: Copying comment by @gthurstonnz

>Line 21: I'd recommend having a sentence that explains the basic premise of why were are doing this and the benefits are from it. Then explain what that the rules can be viewed at rules.nz which is a tool for developers wanting to reuse our rules within their own work.

Our rules can be viewed at <https://rules.nz>

### Birth of a Child (or welcoming a child into your whanau)

> Encoding entitlements related to the birth of a child (or welcoming a child into your whanau) into re-usable open legislation as code. This component was then used within [Smart Start - financial help](https://smartstart.services.govt.nz/financial-help).

Author: Copying comment by @gthurstonnz

> Line 26: I liked how you used an example of how we applied the rules. That would be great for the rest of them too

#### Child related entitlements encoded

Social Security Act 2018

* Accommodation Supplement
* Childcare Subsidy
* Child Disability Allowance
* Community Services Card
* Home Help
* Job Seeker Support
* Orphan's Benefit
* Sole Parent Support
* Supported Living Payment
* Unsupported Child
* Young Parent Payment

New Zealand Superannuation and Retirement Income Act 2001

* Superannuation

Income Tax Act 2007

* Working for Families (various)
* Best Start

Parental Leave and Employment Protection Act 1987

* Paid Parental Leave

### Citizenship by Grant

> A complex formula from within the Citizenship Act, used to calculate if an applicant has been in the country enough days to satisfy a clause. This was encoded into OpenFisca, along with some related sections of Immigration Act.

Author: Copying comment by @gthurstonnz

> Line 59: You might want to explain what OpenFisca is? Its the first time its referenced so a simple explaination about what this and how to use it would be helpful.

### Accident Compensation

> Experiment understanding clauses of the Accident Compensation Act, where descretion is applied, and if these can be encoded. The sections related to Loss of Potential Earnings, and Rehabilitation were encoded into Open Fisca.

Author: Copying comment by @gthurstonnz

> Line 63: Suggest a change to the firest sentence: With ACC and MBIE we explored the entitlement clauses in the Accident Compensation Act to see where descretion is applied, then looked to see how these could be encoded.

> ## Conference talk at Pycon 2019 in Sydney Australia

Author: Copying comment by @gthurstonnz

> Line 65: Suggest we call this section 'Media'. Then we can add in links to the rules blogs on digital.govt.nz and other media too.

> ## Conference talk at Pycon 2019 in Sydney Australia

Author: ```suggestion
## Media

### Conference talk at Pycon 2019 in Sydney Australia
```
